src/Link.php

```diff
@@ -63,6 +63,20 @@ public static function getSectorizedDetails(): array
         return ['config', self::class];
     }
 
+    public static function getLogServiceName(): string
+    {
+        return 'setup';
+    }
+
+    public static function getPostFormAction(string $form_action): ?string
+    {
+        if ($form_action === 'add') {
+            return 'item';
+        }
+
+        return parent::getPostFormAction($form_action);
+    }
```

Comment:
There is no reason for this class to have a different behaviour here. The `item` case here would redirect to the same URL that the URL used in the `if ($result && $form_action === 'add' && $_SESSION['glpibackcreated']) {` case.
You should remove this specific behaviour.